File: ClientApp/src/app/models/OrdersAdditionals.ts
import { Additional } from "./Additional";
import { Order } from "./Order";

export class OrdersAdditionals {

    constructor(order?: Order, additional?: Additional, quantity?: number){
        if(order){
            this.order = order;
        }
        if(additional){
            this.additional_Id = additional.additional_Id;
            this.additional = additional;
        }
        this.quantity = quantity ? quantity : 1;
    }

    id: number;
    order_Id: number;
    additional_Id: number;
    quantity: number;
    order: Order;
    additional: Additional;
    details?: string;
}

File: Models/OrdersAdditionals.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Bakery.Models
{
    public class OrdersAdditionals
    {
        [Key]
        public int Id { get; set; }
        public int Order_Id { get; set; }
        public int Additional_Id { get; set; }
        public int Quantity { get; set; }
        public string Details { get; set; }
        [ForeignKey("Order_Id")]
        public virtual Order Order { get; set; }
        [ForeignKey("Additional_Id")]
        public virtual Additional Additional { get; set; }
    }
}
